settings.gradle

```diff
@@ -20,11 +20,15 @@ plugins {
 
 develocity {
     buildScan {
-        publishing.onlyIf {
-            providers.environmentVariable("CI").isPresent() ||
-                    providers.systemProperty("CI").isPresent() ||
-                    providers.gradleProperty("CI").isPresent()
+        def isCi = providers.environmentVariable("CI").isPresent() ||
+                providers.systemProperty("CI").isPresent() ||
+                providers.gradleProperty("CI").isPresent()
+
+        if (isCi) {
+            publishing.onlyIf { true }
+            uploadInBackground = false
         }
+
```

Comment:
The old code triggered this warning:
```
WARNING: Error invoking build scan publishingCondition action
        Cannot reference a Gradle script object from a Groovy closure as these are not supported with the configuration cache.
```
The new code was closer to a snippet that Gradle provided in one of their samples.